.github/workflows/java.yml

```
@@ -282,18 +282,27 @@ jobs:
         steps:
             - name: Install git and Java
               run: |
-                  yum update -y
-                  yum install -y git tar java-${{ matrix.java }}-amazon-corretto-devel.x86_64
-                  # Set JAVA_HOME to use the installed JDK
-                  export JAVA_HOME=/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64
-                  echo "JAVA_HOME=/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64" >> $GITHUB_ENV
-                  echo "/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64/bin" >> $GITHUB_PATH
                   # Set environment variable to indicate container environment for Gradle
                   echo "GLIDE_CONTAINER_BUILD=true" >> $GITHUB_ENV
-                  # Create gradle user home and disable auto-download
-                  mkdir -p ~/.gradle
-                  echo "org.gradle.java.installations.auto-download=false" >> ~/.gradle/gradle.properties
-                  echo IMAGE=amazonlinux:latest | sed -r 's/:/-/g' >> $GITHUB_ENV
+                  if [[ "${{ matrix.host.OS }}" == "amazon-linux" ]]; then
+                    yum update -y
+                    yum install -y git tar java-${{ matrix.java }}-amazon-corretto-devel.x86_64
+                    # Set JAVA_HOME to use the installed JDK
+                    export JAVA_HOME=/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64
+                    echo "JAVA_HOME=/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64" >> $GITHUB_ENV
+                    echo "/usr/lib/jvm/java-${{ matrix.java }}-amazon-corretto.x86_64/bin" >> $GITHUB_PATH
+                    # Create gradle user home and disable auto-download
+                    mkdir -p ~/.gradle
+                    echo "org.gradle.java.installations.auto-download=false" >> ~/.gradle/gradle.properties
+                    echo IMAGE=amazonlinux:latest | sed -r 's/:/-/g' >> $GITHUB_ENV
+                  elif [[ "${{ matrix.host.TARGET }}" == "x86_64-unknown-linux-musl" ]]; then
+                    apk add openjdk${{matrix.java}} git bash
+                    apk add openjdk11
```

```suggestion
                    apk add openjdk${{matrix.java}} openjdk11 git bash
```
BTW why we need java 11?